### ISSUE TITLE
Redirect to login form when FormAuth is enabled

### ISSUE THREAD

**Issue by frebib** [nzbget/nzbget#535]
Opening a pre-loaded page in a cached browser tab can cause 401 errors requiring a page refresh. It would feel much more natural to have the page automatically redirect instead of having to refresh. Does this sound like a good idea?

**Comment by hugbug** [comment]
I think I don't understand what you mean. How to reproduce this?

**Comment by frebib** [comment]
Particularly on Android, where cached pages are later reloaded after the browser has been restarted, this happens:
![https://i.imgur.com/zCJr7ka.png](https://i.imgur.com/zCJr7ka.png)

I assume this is caused by the page attempting to contact the `/jsonrpc/*` endpoints which return 401 due to session expiration (does NZBGet track sessions with FormAuth?)

**Comment by frebib** [comment]
Apologies for the expired image link, I've updated it

**Comment by frebib** [comment]
The kind of behaviour that I am proposing is something to the effect of a redirection to the login page, when a `401 Unauthorized` error occurs, around this area of the client code: https://github.com/nzbget/nzbget/blob/dcdc41ca9ad923a26015de6800a53c7d44f2ace8/webui/util.js#L639-L646

**Comment by allanak** [comment]
I also get a 401 now when attempting to login with form auth. Switch back to basic has resolved this for now.

**Comment by hugbug** [comment]
@allanak: refreshing page doesn't help?

**Comment by allanak** [comment]
Nope. For clarity, this is directly from the login page rather than losing a session while I'm in the app already. I just realized what's happening here though. Some kind of persistent cookies from organizr (running under the same domain name) are send with the POST to /jsonrpc - incognito browser seemed to fix it. Not sure if this is an organizr issue or an nzbget one :)

**Comment by hugbug** [comment]
If you open nzbget webui in browser directly (not via proxy ui app), then it must be nzbget issue.

**Comment by markheloking** [comment]
This original issue is reproducible on my side on NZBGet 21.0 running in docker, with Chrome for android and Chrome for windows.

Steps to reproduce:

1. Open tap
1. Open nzbget url
1. Login
1. Let login/tab expire (put machine to sleep, come back next day and open stale tab)
1. See 401 status as in https://github.com/nzbget/nzbget/issues/535#issuecomment-387471054

A refresh does provide the login screen and fixes the issue, but an automatic login redirect on 401/stale sessions would be nice.
